src/transformer/utils/Checks.ts

```diff
@@ -3,13 +3,18 @@ export class Checks<TChecks extends { kind: string }[]> {
 
 	find<TKind extends string>(
 		kind: TKind
-	): Utils.FilterChecks<TChecks[number], TKind> | undefined {
+	): FilterChecks<TChecks[number], TKind> | undefined {
 		return this.checks.find((check) => check.kind === kind) as
-			| Utils.FilterChecks<TChecks[number], TKind>
+			| FilterChecks<TChecks[number], TKind>
 			| undefined;
 	}
 
 	has<TKind extends string>(kind: TKind) {
 		return this.find(kind) !== undefined;
 	}
 }
+
+type FilterChecks<
```

Comment:
I would go ahead and export it, even though it's not necessary.

Reply:
From my experience it's best to export as little as possible.
Because that makes the surface bigger, and there's a bigger change of a breaking change 😅
We can always revisit this later and export it if it's required.